packages/contracts-bedrock/src/cannon/MIPS.sol

```diff
@@ -148,277 +139,63 @@ contract MIPS is ISemver {
                 state := 0x80
             }
 
-            // Load the syscall number from the registers
-            uint32 syscall_no = state.registers[2];
+            // Load the syscall numbers and args from the registers
+            (uint32 syscall_no, uint32 a0, uint32 a1, uint32 a2) = sys.getSyscallArgs(state.registers);
+
             uint32 v0 = 0;
             uint32 v1 = 0;
 
-            // Load the syscall arguments from the registers
-            uint32 a0 = state.registers[4];
-            uint32 a1 = state.registers[5];
-            uint32 a2 = state.registers[6];
-
-            // mmap: Allocates a page from the heap.
-            if (syscall_no == 4090) {
-                uint32 sz = a1;
-                if (sz & 4095 != 0) {
-                    // adjust size to align with page size
-                    sz += 4096 - (sz & 4095);
-                }
-                if (a0 == 0) {
-                    v0 = state.heap;
-                    state.heap += sz;
-                } else {
-                    v0 = a0;
-                }
-            }
-            // brk: Returns a fixed address for the program break at 0x40000000
-            else if (syscall_no == 4045) {
+            if (syscall_no == sys.SYS_MMAP) {
```

Comment:
Not sure if there was a reason we weren't using constants here before?

Reply:
No good reason. This is better.